core/threshold/src/execution/online/preprocessing/redis/mod.rs

```diff
@@ -350,6 +351,35 @@ pub struct RedisPreprocessing<R: Ring> {
     _phantom: PhantomData<R>,
 }
 
+impl<R: Ring> Drop for RedisPreprocessing<R> {
+    // Custom drop implementation to make sure there's no dangling
+    // correlated randomness stored in Redis if there are no more references
+    // to it in rust.
+    // NOTE: Ideally we'd Zeroize it but afaict Redis doesn't propose such functionality
+    fn drop(&mut self) {
```

Comment:
I think this drop should be async, otherwise it will block the tokio thread for a long time if there are many keys.

probably the stuff around `con.del` should go into a `tokio::spawn`, I'm not sure if there's a better way